src/test/java/edu/ucla/library/iiif/auth/utils/TestUtils.java

```diff
@@ -0,0 +1,77 @@
+
+package edu.ucla.library.iiif.auth.utils;
+
+import static info.freelibrary.util.Constants.SPACE;
+
+import java.security.SecureRandom;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import edu.ucla.library.iiif.auth.Config;
+import edu.ucla.library.iiif.auth.CookieNames;
+import edu.ucla.library.iiif.auth.MessageCodes;
+import edu.ucla.library.iiif.auth.services.AccessCookieServiceImpl;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * Utilities to assist with testing.
+ */
+@SuppressWarnings({ "PMD.CommentSize", "checkstyle:lineLengthChecker" })
+public final class TestUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestUtils.class, MessageCodes.BUNDLE);
+
+    private TestUtils() {
+    }
+
+    /**
+     * Gets a pair of Sinai cookies by mocking the cookie generation code of the Sinai application.
+     *
+     * @param aConfig The application configuration
+     * @param aLocalDate The date value to put inside the cookie
+     * @return A tuple of size 2 whose first element is a {@link CookieNames#SINAI_CIPHERTEXT} cookie and whose second
+     *         is a {@link CookieNames#SINAI_IV} cookie
+     * @throws Exception If there is an issue generating the cookies
+     * @see <a href=
+     *      "https://github.com/UCLALibrary/sinaimanuscripts/blob/44cbbd9bf508c32b742f1617205a679edf77603e/app/controllers/application_controller.rb#L98-L103">How
+     *      the Sinai application encodes cookies</a>
+     */
+    public static Tuple getMockSinaiCookies(final JsonObject aConfig, final LocalDate aLocalDate) throws Exception {
+        final String clearTextPrefix = aConfig.getString(Config.SINAI_COOKIE_VALID_PREFIX);
+        final String clearTextSuffix =
+                aLocalDate.format(DateTimeFormatter.ofPattern(AccessCookieServiceImpl.SINAI_COOKIE_DATE_FORMAT));
+        final String clearText = String.join(SPACE, clearTextPrefix, clearTextSuffix);
+
+        final Cipher cipher;
+        final SecretKey key;
+        final byte[] cipherText;
+        final String sinaiAuthenticated3Day;
+        final String initializationVector;
+
+        cipher = Cipher.getInstance(AccessCookieServiceImpl.CIPHER_TRANSFORMATION);
```

Comment:
Couldn't you initialize these two variables above, and just leave cipherText and below uninitialized until after the cipher.init()?

Reply:
Okay, done